conjure-macros/src/endpoints.rs

```diff
@@ -0,0 +1,705 @@
+use crate::path::{self, PathComponent};
+use crate::{Asyncness, Errors};
+use heck::ToUpperCamelCase;
+use proc_macro2::{Ident, TokenStream};
+use quote::quote;
+use structmeta::StructMeta;
+use syn::{
+    parse_macro_input, Error, FnArg, ItemTrait, LitStr, Meta, Pat, PatType, TraitItem, TraitItemFn,
+    Type, Visibility,
+};
+
+pub fn generate(
+    _attr: proc_macro::TokenStream,
+    item: proc_macro::TokenStream,
+) -> proc_macro::TokenStream {
+    let mut item = parse_macro_input!(item as ItemTrait);
+    let service = Service::new(&item);
+    strip_trait(&mut item);
```

Comment:
We now unconditionally strip helper attributes so that an error resolving the trait doesn't also cascade into a bunch of errors about unknown attributes. We'll want this in the client code as well.